core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -78,6 +78,14 @@ private CatalogProperties() {}
 
   public static final boolean IO_MANIFEST_CACHE_ENABLED_DEFAULT = false;
 
+  /**
+   * Controls whether engines using a REST Catalog should delegate the drop table purge requests to the Catalog.
+   * Defaults to false, allowing the engine to use its own implementation for purging.
+   */
+  public static final String REST_CATALOG_PURGE = "rest.catalog-purge";
```

Comment:
Isn't the option whether to delegate to the catalog, not specifically REST? Why is the property specific to REST catalog?